alica_engine/src/engine/AlicaEngine.cpp

```diff
@@ -86,6 +89,7 @@ bool AlicaEngine::init(AlicaCreators& creatorCtx)
     _stepCalled = false;
     bool everythingWorked = true;
     everythingWorked &= _behaviourPool.init(*creatorCtx.behaviourCreator);
+    everythingWorked &= _planPool.init(*creatorCtx.planCreator);
```

Comment:
`&=` is a bitwise operator. You are looking for `everythingWorked = everythingWorked && xxx`